src/fondant/executor.py

```diff
@@ -54,13 +76,45 @@ def __init__(
         self.user_arguments = user_arguments
         self.input_partition_rows = input_partition_rows
 
+        if client_kwargs is None:
+            client_kwargs = {
+                "processes": True,
+                "n_workers": os.cpu_count(),
+                "thread_per_worker": 1,
+            }
+
+        if cluster_type == "local":
+            logger.info(f"Initialize local dask cluster with arguments {client_kwargs}")
+
+            # Additional dask configuration have to be set before initialising the client
+            # worker.daemon is set to false because creating a worker process in daemon
+            # mode is not possible in our docker container setup.
+            dask.config.set({"distributed.worker.daemon": False})
+            dask.config.set({"dataframe.convert-string": False})
+
+            local_cluster = LocalCluster(**client_kwargs)
+            self.client = Client(local_cluster)
+
+        elif cluster_type == "distributed":
+            msg = "The usage of the Dask distributed client is not supported yet."
+            raise NotImplementedError(msg)
+        else:
+            logger.info(
+                f"We currently do not support {cluster_type}. "
+                f"Our supported options are limited to 'local' and 'distributed'. "
+                f"Dask local mode will be used for further executions.",
+            )
+            self.client = None
+
     @classmethod
     def from_args(cls) -> "Executor":
         """Create an executor from a passed argument containing the specification as a dict."""
         parser = argparse.ArgumentParser()
         parser.add_argument("--component_spec", type=json.loads)
         parser.add_argument("--cache", type=ast.literal_eval)
         parser.add_argument("--input_partition_rows", type=validate_partition_number)
+        parser.add_argument("--cluster_type", type=ast.literal_eval)
+        parser.add_argument("--client_kwargs", type=json.loads)
```

Comment:
This might fail is `client_kwargs` is none, look at the function `validate_partition_number` to see how this is handled. 

FYI there is some cleanup to do in the whole argument parsing stuff so feel free to give suggestions